pkgs/racket-doc/scribblings/reference/eval-model.scrbl

```diff
@@ -597,62 +615,62 @@ within modules. For example, @racket[define] within @racket[begin-for-syntax] cr
-internal effects is not detectable outside of the Racket program.
+contained in the box. The modified box is not observable outside
+Racket, so the effect is said to be ``internal.'' By definition,
+internal effects are not detectable outside the Racket program.
 
-External effects are exemplified by input/output (or I/O). I/O is the
+External effects are exemplified by input/output (I/O). I/O is the
 action of a function such as @racket[tcp-connect], which communicates
-with the operating system to send network packets outside of the
+with the operating system to send network packets outside the
 machine running Racket. The transmission of these packets is
-observable outside of Racket, in particular by the receiver computer
+observable outside Racket, in particular by the receiving computer
 or any routers in between. External effects exist to be detectable
-outside of the Racket program and are often detectable using physical
+outside the Racket program and are often detectable using physical
 processes.
 
 An effect is @deftech{discarded} when it is no longer detectable. For
-instance, a mutation of a box from @racket[3] to @racket[4] would be
-discarded if it ceases to be detectable that it was ever changed, and
+instance, the mutation of a box from @racket[3] to @racket[4] is
+discarded when it ceases to be detectable that it was ever changed and
 thus would still contain @racket[3]. Because external effects are
-intrinsically observable outside of Racket, they are irreversible and
+intrinsically observable outside Racket, they are irreversible and
 cannot be discarded.
 
 Thus, The Separate Compilation Guarantee only concerns effects like
-mutation, because they are exclusively effects "on the Racket runtime
-system" and not "on the physical universe".
+mutation, because they are exclusively effects ``on the Racket runtime
+system'' and not ``on the physical universe.''
 
 There are many things a Racket program can do that appear to be
-internal effects, but are actually external effects. For instance,
+internal effects but are actually external effects. For instance,
 @racket[bytes-set!] is typically an internal effect, except when the
-bytes were created by @racket[make-shared-bytes] which is allocated in
-space observable by other processes. Thus, effects which modify them
-are not discardable, so @racket[bytes-set!], in this case, is an
+bytes are created by @racket[make-shared-bytes], which allocates in
+space observable by other processes. Thus, effects which modify those
+bytes are not discardable, so @racket[bytes-set!], in this case, has an
 external effect.
 
 The opposite is also true: some things which appear to be external are
 actually internal. For instance, if a Racket program starts multiple
 threads and uses mutation to communicate between them, that mutation
 is purely internal, because Racket's threads are defined entirely
-internally.
+internally. They are not related to operating system threads.
```

Comment:
I think this would read better as a parenthetical at the end of the preceding sentence.

Reply:
Agreed.